core/templates/dev/head/domain/topic/TopicRightsObjectFactory.js

```diff
@@ -19,10 +19,10 @@
 
 oppia.factory('TopicRightsObjectFactory', [
   function() {
-    var TopicRights = function(topicRightsObject) {
-      this._published = topicRightsObject.is_published;
-      this._canPublishTopic = topicRightsObject.can_publish_topic;
-      this._canEditTopic = topicRightsObject.can_edit_topic;
+    var TopicRights = function(topicRightsBackendObject) {
```

Comment:
No, still don't do this -- it's fragile especially if the backend dict changes. Pass each arg separately instead. Let createFromBackendDict parse and interpret the backend object, the constructor shouldn't have to do this.

(Also FYI I think we would call it a backend dict rather than a backend object. I know object is the correct term in JS but that word is pretty generic/overloaded, so that's why we refer to dicts even in the frontend.)

Reply:
Ok, so create would have the backend dict as the argument, and it will split and give the separate fields to the constructor?

Reply:
Yep! (Or, if you're doing that, make sure that "create" is called "createFromBackendDict".)

Reply:
Ok.